float64[] vector
bool write_to_device
---
bool success